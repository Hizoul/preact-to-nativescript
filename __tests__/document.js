import * as preactToNativeScript from "../index"

const h = preactToNativeScript.Preact.h

test("Test Element insertBefore / replaceWith", () => {
  const rendered = preactToNativeScript.render(
    h("page", null, [
      h("scrollView", null, [
        h("stackLayout", null, [
          h("label", {text: "myText"})
        ])
      ])
    ])
  )
  const newChild = preactToNativeScript.render(h("label", {text: "myText"}))
  const newChild2 = preactToNativeScript.render(h("label", {text: "myText"}))
  const newChild3 = preactToNativeScript.render(h("label", {text: "myText"}))
  expect(rendered.childNodes).toMatchSnapshot("children after regular render")
  rendered.replaceChild(newChild, rendered.childNodes[0])
  expect(rendered.childNodes).toMatchSnapshot("children after replace child")
  rendered.insertBefore(newChild2, null)
  expect(rendered.childNodes).toMatchSnapshot("children after insertBefore without ref")
  rendered.insertBefore(newChild3, newChild2)
  expect(rendered.childNodes).toMatchSnapshot("children after insertBefore with valid ref")
  rendered.text = "mytext"
  rendered.appendChild({nodeValue: "myNewText", splitText: ""})
  expect(rendered.text).toBe("myNewText")
  expect(rendered.asdf).toBe(undefined)
  rendered.setAttributeNS(null, "asdf", "myAsdfVal")
  expect(rendered.asdf).toBe("myAsdfVal")
  expect(rendered.asdf).toBe(rendered.getAttributeNS(null, "asdf"))
  rendered.removeAttributeNS(null, "asdf")
  expect(rendered.asdf).toBe(undefined)
  expect(rendered.bla).toBe(undefined)
  rendered.setAttribute("bla", "blu")
  expect(rendered.bla).toBe("blu")
  expect(rendered.bla).toBe(rendered.getAttribute("bla"))
  rendered.removeAttribute("bla")
  expect(rendered.bla).toBe(null)
})
