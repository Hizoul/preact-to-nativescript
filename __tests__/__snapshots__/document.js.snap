// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Element insertBefore / replaceWith: children after insertBefore with valid ref 1`] = `
Array [
  Label {
    "__n": "label",
    "__preactattr_": Object {
      "text": "myText",
    },
    "_eventListener": Array [],
    "attributes": Array [],
    "childNodes": Array [],
    "firstLoad": true,
    "loaded": false,
    "mergeInto": undefined,
    "nodeName": "LABEL",
    "nodeType": 1,
    "parentNode": Page {
      "__n": "page",
      "__preactattr_": Object {},
      "_eventListener": Array [],
      "attributes": Array [],
      "childNodes": [Circular],
      "content": Label {
        "__n": "label",
        "__preactattr_": Object {
          "text": "myText",
        },
        "_eventListener": Array [],
        "attributes": Array [],
        "childNodes": Array [],
        "firstLoad": true,
        "loaded": false,
        "mergeInto": undefined,
        "nodeName": "LABEL",
        "nodeType": 1,
        "parentNode": [Circular],
        "renderedComponent": VNode {
          "attributes": Object {
            "text": "myText",
          },
          "children": Array [],
          "key": undefined,
          "nodeName": "label",
        },
        "renderedInto": [Circular],
        "set": [Function],
        "text": "myText",
      },
      "firstLoad": true,
      "loaded": false,
      "mergeInto": undefined,
      "nodeName": "PAGE",
      "nodeType": 1,
      "parentNode": Object {
        "appendChild": [Function],
        "attributes": Array [],
        "childNodes": Array [
          [Circular],
        ],
        "mergeInto": undefined,
        "nodeName": "artificalParent",
        "nodeType": "artificalParent",
        "remove": [Function],
        "removeChild": [Function],
        "renderedComponent": VNode {
          "attributes": undefined,
          "children": Array [
            VNode {
              "attributes": undefined,
              "children": Array [
                VNode {
                  "attributes": undefined,
                  "children": Array [
                    VNode {
                      "attributes": Object {
                        "text": "myText",
                      },
                      "children": Array [],
                      "key": undefined,
                      "nodeName": "label",
                    },
                  ],
                  "key": undefined,
                  "nodeName": "stackLayout",
                },
              ],
              "key": undefined,
              "nodeName": "scrollView",
            },
          ],
          "key": undefined,
          "nodeName": "page",
        },
        "renderedInto": null,
      },
      "renderedComponent": VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              VNode {
                "attributes": undefined,
                "children": Array [
                  VNode {
                    "attributes": Object {
                      "text": "myText",
                    },
                    "children": Array [],
                    "key": undefined,
                    "nodeName": "label",
                  },
                ],
                "key": undefined,
                "nodeName": "stackLayout",
              },
            ],
            "key": undefined,
            "nodeName": "scrollView",
          },
        ],
        "key": undefined,
        "nodeName": "page",
      },
      "renderedInto": [Circular],
      "set": [Function],
    },
    "renderedComponent": VNode {
      "attributes": Object {
        "text": "myText",
      },
      "children": Array [],
      "key": undefined,
      "nodeName": "label",
    },
    "renderedInto": [Circular],
    "set": [Function],
    "text": "myText",
  },
  Label {
    "__n": "label",
    "__preactattr_": Object {
      "text": "myText",
    },
    "_eventListener": Array [],
    "attributes": Array [],
    "childNodes": Array [],
    "firstLoad": true,
    "loaded": false,
    "mergeInto": undefined,
    "nodeName": "LABEL",
    "nodeType": 1,
    "parentNode": Page {
      "__n": "page",
      "__preactattr_": Object {},
      "_eventListener": Array [],
      "attributes": Array [],
      "childNodes": [Circular],
      "content": [Circular],
      "firstLoad": true,
      "loaded": false,
      "mergeInto": undefined,
      "nodeName": "PAGE",
      "nodeType": 1,
      "parentNode": Object {
        "appendChild": [Function],
        "attributes": Array [],
        "childNodes": Array [
          [Circular],
        ],
        "mergeInto": undefined,
        "nodeName": "artificalParent",
        "nodeType": "artificalParent",
        "remove": [Function],
        "removeChild": [Function],
        "renderedComponent": VNode {
          "attributes": undefined,
          "children": Array [
            VNode {
              "attributes": undefined,
              "children": Array [
                VNode {
                  "attributes": undefined,
                  "children": Array [
                    VNode {
                      "attributes": Object {
                        "text": "myText",
                      },
                      "children": Array [],
                      "key": undefined,
                      "nodeName": "label",
                    },
                  ],
                  "key": undefined,
                  "nodeName": "stackLayout",
                },
              ],
              "key": undefined,
              "nodeName": "scrollView",
            },
          ],
          "key": undefined,
          "nodeName": "page",
        },
        "renderedInto": null,
      },
      "renderedComponent": VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              VNode {
                "attributes": undefined,
                "children": Array [
                  VNode {
                    "attributes": Object {
                      "text": "myText",
                    },
                    "children": Array [],
                    "key": undefined,
                    "nodeName": "label",
                  },
                ],
                "key": undefined,
                "nodeName": "stackLayout",
              },
            ],
            "key": undefined,
            "nodeName": "scrollView",
          },
        ],
        "key": undefined,
        "nodeName": "page",
      },
      "renderedInto": [Circular],
      "set": [Function],
    },
    "renderedComponent": VNode {
      "attributes": Object {
        "text": "myText",
      },
      "children": Array [],
      "key": undefined,
      "nodeName": "label",
    },
    "renderedInto": [Circular],
    "set": [Function],
    "text": "myText",
  },
  Label {
    "__n": "label",
    "__preactattr_": Object {
      "text": "myText",
    },
    "_eventListener": Array [],
    "attributes": Array [],
    "childNodes": Array [],
    "firstLoad": true,
    "loaded": false,
    "mergeInto": undefined,
    "nodeName": "LABEL",
    "nodeType": 1,
    "parentNode": Page {
      "__n": "page",
      "__preactattr_": Object {},
      "_eventListener": Array [],
      "attributes": Array [],
      "childNodes": [Circular],
      "content": Label {
        "__n": "label",
        "__preactattr_": Object {
          "text": "myText",
        },
        "_eventListener": Array [],
        "attributes": Array [],
        "childNodes": Array [],
        "firstLoad": true,
        "loaded": false,
        "mergeInto": undefined,
        "nodeName": "LABEL",
        "nodeType": 1,
        "parentNode": [Circular],
        "renderedComponent": VNode {
          "attributes": Object {
            "text": "myText",
          },
          "children": Array [],
          "key": undefined,
          "nodeName": "label",
        },
        "renderedInto": [Circular],
        "set": [Function],
        "text": "myText",
      },
      "firstLoad": true,
      "loaded": false,
      "mergeInto": undefined,
      "nodeName": "PAGE",
      "nodeType": 1,
      "parentNode": Object {
        "appendChild": [Function],
        "attributes": Array [],
        "childNodes": Array [
          [Circular],
        ],
        "mergeInto": undefined,
        "nodeName": "artificalParent",
        "nodeType": "artificalParent",
        "remove": [Function],
        "removeChild": [Function],
        "renderedComponent": VNode {
          "attributes": undefined,
          "children": Array [
            VNode {
              "attributes": undefined,
              "children": Array [
                VNode {
                  "attributes": undefined,
                  "children": Array [
                    VNode {
                      "attributes": Object {
                        "text": "myText",
                      },
                      "children": Array [],
                      "key": undefined,
                      "nodeName": "label",
                    },
                  ],
                  "key": undefined,
                  "nodeName": "stackLayout",
                },
              ],
              "key": undefined,
              "nodeName": "scrollView",
            },
          ],
          "key": undefined,
          "nodeName": "page",
        },
        "renderedInto": null,
      },
      "renderedComponent": VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              VNode {
                "attributes": undefined,
                "children": Array [
                  VNode {
                    "attributes": Object {
                      "text": "myText",
                    },
                    "children": Array [],
                    "key": undefined,
                    "nodeName": "label",
                  },
                ],
                "key": undefined,
                "nodeName": "stackLayout",
              },
            ],
            "key": undefined,
            "nodeName": "scrollView",
          },
        ],
        "key": undefined,
        "nodeName": "page",
      },
      "renderedInto": [Circular],
      "set": [Function],
    },
    "renderedComponent": VNode {
      "attributes": Object {
        "text": "myText",
      },
      "children": Array [],
      "key": undefined,
      "nodeName": "label",
    },
    "renderedInto": [Circular],
    "set": [Function],
    "text": "myText",
  },
]
`;

exports[`Test Element insertBefore / replaceWith: children after insertBefore without ref 1`] = `
Array [
  Label {
    "__n": "label",
    "__preactattr_": Object {
      "text": "myText",
    },
    "_eventListener": Array [],
    "attributes": Array [],
    "childNodes": Array [],
    "firstLoad": true,
    "loaded": false,
    "mergeInto": undefined,
    "nodeName": "LABEL",
    "nodeType": 1,
    "parentNode": Page {
      "__n": "page",
      "__preactattr_": Object {},
      "_eventListener": Array [],
      "attributes": Array [],
      "childNodes": [Circular],
      "content": Label {
        "__n": "label",
        "__preactattr_": Object {
          "text": "myText",
        },
        "_eventListener": Array [],
        "attributes": Array [],
        "childNodes": Array [],
        "firstLoad": true,
        "loaded": false,
        "mergeInto": undefined,
        "nodeName": "LABEL",
        "nodeType": 1,
        "parentNode": [Circular],
        "renderedComponent": VNode {
          "attributes": Object {
            "text": "myText",
          },
          "children": Array [],
          "key": undefined,
          "nodeName": "label",
        },
        "renderedInto": [Circular],
        "set": [Function],
        "text": "myText",
      },
      "firstLoad": true,
      "loaded": false,
      "mergeInto": undefined,
      "nodeName": "PAGE",
      "nodeType": 1,
      "parentNode": Object {
        "appendChild": [Function],
        "attributes": Array [],
        "childNodes": Array [
          [Circular],
        ],
        "mergeInto": undefined,
        "nodeName": "artificalParent",
        "nodeType": "artificalParent",
        "remove": [Function],
        "removeChild": [Function],
        "renderedComponent": VNode {
          "attributes": undefined,
          "children": Array [
            VNode {
              "attributes": undefined,
              "children": Array [
                VNode {
                  "attributes": undefined,
                  "children": Array [
                    VNode {
                      "attributes": Object {
                        "text": "myText",
                      },
                      "children": Array [],
                      "key": undefined,
                      "nodeName": "label",
                    },
                  ],
                  "key": undefined,
                  "nodeName": "stackLayout",
                },
              ],
              "key": undefined,
              "nodeName": "scrollView",
            },
          ],
          "key": undefined,
          "nodeName": "page",
        },
        "renderedInto": null,
      },
      "renderedComponent": VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              VNode {
                "attributes": undefined,
                "children": Array [
                  VNode {
                    "attributes": Object {
                      "text": "myText",
                    },
                    "children": Array [],
                    "key": undefined,
                    "nodeName": "label",
                  },
                ],
                "key": undefined,
                "nodeName": "stackLayout",
              },
            ],
            "key": undefined,
            "nodeName": "scrollView",
          },
        ],
        "key": undefined,
        "nodeName": "page",
      },
      "renderedInto": [Circular],
      "set": [Function],
    },
    "renderedComponent": VNode {
      "attributes": Object {
        "text": "myText",
      },
      "children": Array [],
      "key": undefined,
      "nodeName": "label",
    },
    "renderedInto": [Circular],
    "set": [Function],
    "text": "myText",
  },
  Label {
    "__n": "label",
    "__preactattr_": Object {
      "text": "myText",
    },
    "_eventListener": Array [],
    "attributes": Array [],
    "childNodes": Array [],
    "firstLoad": true,
    "loaded": false,
    "mergeInto": undefined,
    "nodeName": "LABEL",
    "nodeType": 1,
    "parentNode": Page {
      "__n": "page",
      "__preactattr_": Object {},
      "_eventListener": Array [],
      "attributes": Array [],
      "childNodes": [Circular],
      "content": [Circular],
      "firstLoad": true,
      "loaded": false,
      "mergeInto": undefined,
      "nodeName": "PAGE",
      "nodeType": 1,
      "parentNode": Object {
        "appendChild": [Function],
        "attributes": Array [],
        "childNodes": Array [
          [Circular],
        ],
        "mergeInto": undefined,
        "nodeName": "artificalParent",
        "nodeType": "artificalParent",
        "remove": [Function],
        "removeChild": [Function],
        "renderedComponent": VNode {
          "attributes": undefined,
          "children": Array [
            VNode {
              "attributes": undefined,
              "children": Array [
                VNode {
                  "attributes": undefined,
                  "children": Array [
                    VNode {
                      "attributes": Object {
                        "text": "myText",
                      },
                      "children": Array [],
                      "key": undefined,
                      "nodeName": "label",
                    },
                  ],
                  "key": undefined,
                  "nodeName": "stackLayout",
                },
              ],
              "key": undefined,
              "nodeName": "scrollView",
            },
          ],
          "key": undefined,
          "nodeName": "page",
        },
        "renderedInto": null,
      },
      "renderedComponent": VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              VNode {
                "attributes": undefined,
                "children": Array [
                  VNode {
                    "attributes": Object {
                      "text": "myText",
                    },
                    "children": Array [],
                    "key": undefined,
                    "nodeName": "label",
                  },
                ],
                "key": undefined,
                "nodeName": "stackLayout",
              },
            ],
            "key": undefined,
            "nodeName": "scrollView",
          },
        ],
        "key": undefined,
        "nodeName": "page",
      },
      "renderedInto": [Circular],
      "set": [Function],
    },
    "renderedComponent": VNode {
      "attributes": Object {
        "text": "myText",
      },
      "children": Array [],
      "key": undefined,
      "nodeName": "label",
    },
    "renderedInto": [Circular],
    "set": [Function],
    "text": "myText",
  },
]
`;

exports[`Test Element insertBefore / replaceWith: children after regular render 1`] = `
Array [
  ScrollView {
    "__n": "scrollView",
    "__preactattr_": Object {},
    "_eventListener": Array [],
    "attributes": Array [],
    "childNodes": Array [
      StackLayout {
        "__n": "stackLayout",
        "__preactattr_": Object {},
        "_eventListener": Array [],
        "attributes": Array [],
        "childNodes": Array [
          Label {
            "__n": "label",
            "__preactattr_": Object {
              "text": "myText",
            },
            "_eventListener": Array [],
            "attributes": Array [],
            "childNodes": Array [],
            "firstLoad": true,
            "loaded": false,
            "nodeName": "LABEL",
            "nodeType": 1,
            "parentNode": [Circular],
            "set": [Function],
            "text": "myText",
          },
        ],
        "firstLoad": true,
        "loaded": false,
        "nodeName": "STACKLAYOUT",
        "nodeType": 1,
        "parentNode": [Circular],
        "set": [Function],
      },
    ],
    "content": StackLayout {
      "__n": "stackLayout",
      "__preactattr_": Object {},
      "_eventListener": Array [],
      "attributes": Array [],
      "childNodes": Array [
        Label {
          "__n": "label",
          "__preactattr_": Object {
            "text": "myText",
          },
          "_eventListener": Array [],
          "attributes": Array [],
          "childNodes": Array [],
          "firstLoad": true,
          "loaded": false,
          "nodeName": "LABEL",
          "nodeType": 1,
          "parentNode": [Circular],
          "set": [Function],
          "text": "myText",
        },
      ],
      "firstLoad": true,
      "loaded": false,
      "nodeName": "STACKLAYOUT",
      "nodeType": 1,
      "parentNode": [Circular],
      "set": [Function],
    },
    "firstLoad": true,
    "loaded": false,
    "nodeName": "SCROLLVIEW",
    "nodeType": 1,
    "parentNode": Page {
      "__n": "page",
      "__preactattr_": Object {},
      "_eventListener": Array [],
      "attributes": Array [],
      "childNodes": [Circular],
      "content": [Circular],
      "firstLoad": true,
      "loaded": false,
      "mergeInto": undefined,
      "nodeName": "PAGE",
      "nodeType": 1,
      "parentNode": Object {
        "appendChild": [Function],
        "attributes": Array [],
        "childNodes": Array [
          [Circular],
        ],
        "mergeInto": undefined,
        "nodeName": "artificalParent",
        "nodeType": "artificalParent",
        "remove": [Function],
        "removeChild": [Function],
        "renderedComponent": VNode {
          "attributes": undefined,
          "children": Array [
            VNode {
              "attributes": undefined,
              "children": Array [
                VNode {
                  "attributes": undefined,
                  "children": Array [
                    VNode {
                      "attributes": Object {
                        "text": "myText",
                      },
                      "children": Array [],
                      "key": undefined,
                      "nodeName": "label",
                    },
                  ],
                  "key": undefined,
                  "nodeName": "stackLayout",
                },
              ],
              "key": undefined,
              "nodeName": "scrollView",
            },
          ],
          "key": undefined,
          "nodeName": "page",
        },
        "renderedInto": null,
      },
      "renderedComponent": VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              VNode {
                "attributes": undefined,
                "children": Array [
                  VNode {
                    "attributes": Object {
                      "text": "myText",
                    },
                    "children": Array [],
                    "key": undefined,
                    "nodeName": "label",
                  },
                ],
                "key": undefined,
                "nodeName": "stackLayout",
              },
            ],
            "key": undefined,
            "nodeName": "scrollView",
          },
        ],
        "key": undefined,
        "nodeName": "page",
      },
      "renderedInto": [Circular],
      "set": [Function],
    },
    "set": [Function],
  },
]
`;

exports[`Test Element insertBefore / replaceWith: children after replace child 1`] = `
Array [
  Label {
    "__n": "label",
    "__preactattr_": Object {
      "text": "myText",
    },
    "_eventListener": Array [],
    "attributes": Array [],
    "childNodes": Array [],
    "firstLoad": true,
    "loaded": false,
    "mergeInto": undefined,
    "nodeName": "LABEL",
    "nodeType": 1,
    "parentNode": Page {
      "__n": "page",
      "__preactattr_": Object {},
      "_eventListener": Array [],
      "attributes": Array [],
      "childNodes": [Circular],
      "content": [Circular],
      "firstLoad": true,
      "loaded": false,
      "mergeInto": undefined,
      "nodeName": "PAGE",
      "nodeType": 1,
      "parentNode": Object {
        "appendChild": [Function],
        "attributes": Array [],
        "childNodes": Array [
          [Circular],
        ],
        "mergeInto": undefined,
        "nodeName": "artificalParent",
        "nodeType": "artificalParent",
        "remove": [Function],
        "removeChild": [Function],
        "renderedComponent": VNode {
          "attributes": undefined,
          "children": Array [
            VNode {
              "attributes": undefined,
              "children": Array [
                VNode {
                  "attributes": undefined,
                  "children": Array [
                    VNode {
                      "attributes": Object {
                        "text": "myText",
                      },
                      "children": Array [],
                      "key": undefined,
                      "nodeName": "label",
                    },
                  ],
                  "key": undefined,
                  "nodeName": "stackLayout",
                },
              ],
              "key": undefined,
              "nodeName": "scrollView",
            },
          ],
          "key": undefined,
          "nodeName": "page",
        },
        "renderedInto": null,
      },
      "renderedComponent": VNode {
        "attributes": undefined,
        "children": Array [
          VNode {
            "attributes": undefined,
            "children": Array [
              VNode {
                "attributes": undefined,
                "children": Array [
                  VNode {
                    "attributes": Object {
                      "text": "myText",
                    },
                    "children": Array [],
                    "key": undefined,
                    "nodeName": "label",
                  },
                ],
                "key": undefined,
                "nodeName": "stackLayout",
              },
            ],
            "key": undefined,
            "nodeName": "scrollView",
          },
        ],
        "key": undefined,
        "nodeName": "page",
      },
      "renderedInto": [Circular],
      "set": [Function],
    },
    "renderedComponent": VNode {
      "attributes": Object {
        "text": "myText",
      },
      "children": Array [],
      "key": undefined,
      "nodeName": "label",
    },
    "renderedInto": [Circular],
    "set": [Function],
    "text": "myText",
  },
]
`;
